upcoming: [M3-9403] - Update copies for Linode Interfaces: Linode Details, Account Settings, Firewalls

coliu-akamai opened the pull request:
## Description 📝

Updates copy for various sections of the Linode Interface project

## Changes  🔄
In addition to updating copies,

- fix IP address table hiding 'Add an IP Address' button for config interfaces

Upgrading interfaces:
- add tooltip to chip in LInode Entity details
- Show interfaces preview upon successful dry run
- add beta chip to dialog title
- add return to overview button
- Upon success of actual upgrade - add button to go to linode's network section

Edit Interfaces
- Update banner to info banner (same with Add interface)
- Add ID to drawer title

Account settings Network Interface type
- Move Linode related options ahead of Config options

## Target release date 🗓️
5/6

## Preview 📷

**Include a screenshot or screen recording of the change.**

:lock: Use the [Mask Sensitive Data](https://cloud.linode.com/profile/settings) setting for security.

:bulb: Use `<video src="" />` tag when including recordings in table.

| Before  | After   |
| ------- | ------- |
| ![image](https://github.com/user-attachments/assets/f5fa7f75-a040-43e8-aeeb-ceb973ebac42) | ![image](https://github.com/user-attachments/assets/22fc7767-b5de-416d-a9fa-07481d57f8be) |
| ![image](https://github.com/user-attachments/assets/d46372e7-5e9d-48ef-ad55-a8caf8539d00) | ![image](https://github.com/user-attachments/assets/c4be7a20-af22-475c-9e1f-841550b94c50) |
| ![image](https://github.com/user-attachments/assets/6d81974c-4734-48a0-a479-b1ea1187a4c0) | ![image](https://github.com/user-attachments/assets/adab0ee1-1a7d-4bb0-9db1-455cdd926d20) |
| ![image](https://github.com/user-attachments/assets/a534045b-0321-4d8e-afbf-090152bce7b1) | ![image](https://github.com/user-attachments/assets/dcf39f15-5b1e-43e8-84d9-344238030542) |

## How to test 🧪
- Linode interfaces customer tag and feature flag on

### Verification steps
- confirm copy updated as per spreadsheet/figma file/prototype (linked in ticket) for Linode Details, Account Settings, and Firewalls. For Linode Create flow, see #12129 
- Add IP Address button - confirm this shows up in the IP table for Linodes with legacy interfaces

> [!NOTE]  
> Some copies (especially ones highlighed in blue in the spreadsheet) may be updated soon - they're being worked on 

<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

👀 Doing a self review
❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
🤏 Splitting feature into small PRs
➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
🧪 Providing/improving test coverage
 🔐 Removing all sensitive information from the code and PR description
🚩 Using a feature flag to protect the release
👣 Providing comprehensive reproduction steps
📑 Providing or updating our documentation
🕛 Scheduling a pair reviewing session
📱 Providing mobile support
♿  Providing accessibility support

<br/>

- [ ] I have read and considered all applicable items listed above.

## As an Author, before moving this PR from Draft to Open, I confirmed ✅

- [ ] All unit tests are passing
- [ ] TypeScript compilation succeeded without errors
- [ ] Code passes all linting rules

</details>

bnussman-akamai commented on packages/manager/src/features/Linodes/LinodesDetail/LinodeConfigs/UpgradeInterfaces/DialogContents/ConfigSelectDialogContent.tsx:
```diff
@@ -68,13 +74,20 @@ export const ConfigSelectDialogContent = (
         <Button buttonType="secondary" disabled={isPending} onClick={onClose}>
           Cancel
         </Button>
+        <Button
+          buttonType="outlined"
+          disabled={isPending}
+          onClick={() => setDialogState({ ...initialState })}
+        >
+          Return to Overview
+        </Button>
```
This is so much better! Thanks for adding 🙏

bnussman-akamai commented on packages/manager/src/features/Linodes/LinodesDetail/LinodeConfigs/UpgradeInterfaces/DialogContents/SuccessDialogContent.tsx:
What does the `0` mean here?

Testing the upgrade flow for a Linode that just has a VPC interface with a 1:1 NAT
![Screenshot 2025-04-30 at 12 28 36 PM](https://github.com/user-attachments/assets/7e47b231-e2bb-4b77-97c2-41b8f40e7142)

bnussman-akamai commented on packages/manager/src/features/Linodes/LinodesDetail/LinodeConfigs/UpgradeInterfaces/UpgradeInterfacesDialog.tsx:
Related to @hkhalil-akamai 's question earlier about the fullHeight dialog. 

This is just my opinion, but I think I'd rather have a centered dialog even if it means the content will move a bit on each stage

No action required now on this, but if enough people agree we could make the change.

coliu-akamai replied:
def open to revisiting - lemme cc you in the thread where we had discussed positioning earlier

bnussman-akamai commented on packages/manager/src/features/Linodes/LinodesDetail/LinodeConfigs/UpgradeInterfaces/DialogContents/ErrorDialogContent.tsx:
This might be more of a "bug" at the Notice level rather than with your code, but I think a full width notice would look wayyy better here

![Screenshot 2025-04-30 at 12 34 55 PM](https://github.com/user-attachments/assets/924b5a0d-3860-41a7-b94b-1af1382b7ede)

coliu-akamai replied:
agreed, trying to look into how to get back the full length notice - I know notices outside a paper will take the full length, but I'm having trouble so far - not sure if the Dialog uses an underlying paper 😖

coliu-akamai replied:
having a bit of trouble with this one - will continue poking at it

bnussman-akamai commented on packages/manager/src/features/Linodes/LinodesDetail/LinodeConfigs/UpgradeInterfaces/DialogContents/SuccessDialogContent.tsx:
```diff
@@ -26,22 +31,25 @@ export const SuccessDialogContent = (
   return (
     <Stack gap={2}>
-      <Notice variant="success">
+      <Notice spacingBottom={0} variant="success">
```
Same thing here, I think a full with notice would look wayyyy better

![Screenshot 2025-04-30 at 12 38 19 PM](https://github.com/user-attachments/assets/d5def7ab-8b7b-4674-bb42-b9ad529951ce)

bnussman-akamai replied:
We will figure this out at a later point. We need to discuess with core design people about what the treatment should be

coliu-akamai replied:
(on the bright side for this specific banner.... the newest copy is lengthier and extends the banner to be full width 😆)

bnussman-akamai reviewed (commented):
Looking very good so far. Just gonna do a bit more testing before approving

mjac0bs reviewed (approved):
Lots of little things going on in this PR, but I think things are working as intended from what I tested.

- Verified copy and didn't see any major issues, just noted a couple of places for potential improvement
- Confirmed the Add An IP Address button shows up in the IP table for Linodes with legacy interfaces, but not Linode interfaces
- Beta chip and interface IDs are present where described
- Dry run's new buttons and summary look good

Thanks Connie

mjac0bs commented on packages/ui/src/components/DialogTitle/DialogTitle.tsx:
```diff
@@ -13,11 +13,13 @@ export interface DialogTitleProps {
   subtitle?: string;
   sx?: SxProps<Theme>;
   title: string;
+  titleChip?: JSX.Element;
```
> Maybe instead of titleChip we use titleSuffix or something a bit more generic? This is probably my vote for now ⭐

I would agree with you on the naming!

coliu-akamai replied:
updating to titleSuffix!

coliu-akamai commented on packages/manager/src/features/Linodes/LinodesDetail/LinodeNetworking/LinodeInterfaces/InterfaceSettingsForm.tsx:
will add this in with all the other feedback - ty!

bnussman-akamai commented on packages/manager/src/features/Linodes/LinodesDetail/LinodeNetworking/LinodeInterfaces/InterfaceSettingsForm.tsx:
```diff
@@ -192,7 +192,7 @@ export const InterfaceSettingsForm = (props: Props) => {
               <FormControlLabel
                 checked={field.value ?? false}
                 control={<Toggle />}
-                label="Automatically configure networking"
```
I'm going to check with UX and Daniel on this

`Automatically configure networking` sounds much better to me than `Automatically deposit a networking configuration`

hkhalil-akamai commented on packages/manager/src/features/Account/NetworkInterfaceType.tsx:
```diff
@@ -127,3 +126,51 @@ export const NetworkInterfaceType = () => {
   );
 };
+
+const optionsTooltipText = (
```
The order of these options is different from the copy spreadsheet. Just wanted to point that out in case it wasn't intentional.

coliu-akamai commented:
thanks everyone! addressed feedback + input the most recent copy updates including:
- Automatically deposit a networking configuration -> Enable Network Helper
- Removed the todo comment from SubnetLinodeRow - we may have to update the copy later, but I think what we have should hopefully work

bnussman-akamai commented:
Non blocking small bug: There is no spacing between these buttons.

![Screenshot 2025-04-30 at 5 19 24 PM](https://github.com/user-attachments/assets/0c77eaa7-d68c-41ed-ae89-a4e1d21549cf)

coliu-akamai commented:
ty Banks! I'm gonna sneak that in once the e2e tests finish running 🙏 
and then merge after required checks pass on that commit since this shouldn't e2e results
